arrayfire/array_api/array_object.py

```diff
@@ -0,0 +1,1093 @@
+        return _process_c_function(self, other, backend.get().af_bitand)
+
+    def __ior__(self, other: Union[int, bool, Array], /) -> Array:
+        """
+        Return self |= other.
+        """
+        return _process_c_function(self, other, backend.get().af_bitor)
+
+    def __ixor__(self, other: Union[int, bool, Array], /) -> Array:
+        """
+        Return self ^= other.
+        """
+        return _process_c_function(self, other, backend.get().af_bitxor)
+
+    def __ilshift__(self, other: Union[int, Array], /) -> Array:
+        """
+        Return self <<= other.
+        """
+        return _process_c_function(self, other, backend.get().af_bitshiftl)
+
+    def __irshift__(self, other: Union[int, Array], /) -> Array:
+        """
+        Return self >>= other.
+        """
+        return _process_c_function(self, other, backend.get().af_bitshiftr)
+
+    # Methods
+
+    def __abs__(self) -> Array:
+        # TODO
+        return NotImplemented
+
+    def __array_namespace__(self, *, api_version: Optional[str] = None) -> Any:
+        # TODO
+        return NotImplemented
+
+    def __bool__(self) -> bool:
+        # TODO consider using scalar() and is_scalar()
+        return NotImplemented
+
+    def __complex__(self) -> complex:
+        # TODO
+        return NotImplemented
+
+    def __dlpack__(self, *, stream: Union[None, int, Any] = None):  # type: ignore[no-untyped-def]
+        # TODO implementation and expected return type -> PyCapsule
+        return NotImplemented
+
+    def __dlpack_device__(self) -> Tuple[enum.Enum, int]:
+        # TODO
+        return NotImplemented
+
+    def __float__(self) -> float:
+        # TODO
+        return NotImplemented
+
+    def __getitem__(self, key: Union[int, slice, Tuple[Union[int, slice, ], ...], Array], /) -> Array:
+        """
+        Returns self[key].
+
+        Parameters
+        ----------
+        self : Array
+            Array instance.
+        key : Union[int, slice, Tuple[Union[int, slice, ], ...], Array]
+            Index key.
+
+        Returns
+        -------
+        out : Array
+            An array containing the accessed value(s). The returned array must have the same data type as self.
+        """
+        # TODO
+        # API Specification - key: Union[int, slice, ellipsis, Tuple[Union[int, slice, ellipsis], ...], array].
+        # consider using af.span to replace ellipsis during refactoring
+        out = Array()
+        ndims = self.ndim
+
+        if isinstance(key, Array) and key == af_bool.c_api_value:
+            ndims = 1
+            if count(key) == 0:
+                return out
+
+        safe_call(backend.get().af_index_gen(
+            ctypes.pointer(out.arr), self.arr, c_dim_t(ndims), _get_indices(key).pointer))
+        return out
+
+    def __index__(self) -> int:
+        # TODO
+        return NotImplemented
+
+    def __int__(self) -> int:
+        # TODO
+        return NotImplemented
+
+    def __len__(self) -> int:
```

Comment:
remove, not part of the spec. Would be good to verify all of these and remove extra functions or make them private

Reply:
Let's move/group all non-standard functions into separate region, at bottom? delineate with comments